python_modules/libraries/dagster-aws/dagster_aws/athena/resources.py

```diff
@@ -0,0 +1,263 @@
+import csv
+import io
+import os
+import time
+import uuid
+
+import boto3
+from botocore.stub import Stubber
+from dagster import Field, StringSource, check, resource
+from dagster.seven import urlparse
+
+
+class AthenaError(Exception):
+    pass
+
+
+class AthenaTimeout(AthenaError):
+    pass
+
+
+class AthenaResource:
+    def __init__(self, client, workgroup="primary", polling_interval=5, max_polls=120):
+        self.client = client
+        self.workgroup = workgroup
+        self.max_polls = max_polls
+        self.polling_interval = polling_interval
+
+    def execute_query(self, query, fetch_results=False):
+        """Synchronously execute a single query against Athena. If fetch_results is set to true,
+        will return a list of rows, where each row is a tuple of stringified values,
+        e.g. SELECT 1 will return [("1",)].
+
+        Args:
+            query (str): The query to execute.
+            fetch_results (Optional[bool]): Whether to return the results of executing the query.
+                Defaults to False, in which case the query will be executed without retrieving the
+                results.
+
+        Returns:
+            Optional[List[Tuple[Optional[str], ...]]]: Results of the query, as a list of tuples,
+                when fetch_results is set. Otherwise, return None. All items in the tuple are
+                represented as strings except for empty columns which are represented as None.
+        """
+        check.str_param(query, "query")
+        check.bool_param(fetch_results, "fetch_results")
+        execution_id = self.client.start_query_execution(
+            QueryString=query, WorkGroup=self.workgroup
+        )["QueryExecutionId"]
+        self._poll(execution_id)
+        if fetch_results:
+            return self._results(execution_id)
+
+    def _poll(self, execution_id):
+        retries = self.max_polls
```

Comment:
i think user setting `max_polls = 0` (and `max_polls < 0`) will cause the polling loop (L58-66) to not run and the an AthenaTimeout to be thrown (L69). should we enforce max_polls > 0?